provider/thedoctor.py

```diff
@@ -27,7 +27,8 @@
 
 class TheDoctor (Thread):
     # maximum time to allow a consumer to not successfully poll() before restarting
-    poll_timeout_seconds = 20
+    # this value must be greater than the total amount of time a consumer might retry firing a trigger
+    poll_timeout_seconds = 200
```

Comment:
This is the maximum amount of time between successful polls before the doctor will forcibly restart the consumer. It is a drastic measure that should not happen unless absolutely necessary. Keep in mind that previously, the total retry time was only 10 seconds and the `poll_timeout_seconds` was double that.